### ISSUE TITLE
Multiple NC Servers

### ISSUE THREAD

**Issue by jkalousek** [stefan-niedermann/nextcloud-notes#171]
Hi, is there any chance that you could add support for Multiple NextCloud Servers?

We are using NextCloud in our company, but I'm also using NextCloud at home and it would be nice to be able to switch between them.

**Comment by korelstar** [comment]
Thought about that, too. That shouldn't be that difficult. For switching between accounts, one could use the planned navigation sidebar (see #165) like in the Nextcloud Android app. One had to decide whether only the current active account should be synchronized or all accounts together.

However, I have currently no time for implementing this. Other things are more important for me. But I like the idea.

**Comment by jkalousek** [comment]
Yes, navigation sidebar is sort of standard in android Apps (in my opinion). I would suggest that there could be option for synchronization in each server's setting (synchronize with other/global or manual).

**Comment by stefan-niedermann** [comment]
Agree this would be a useful feature. As @korelstar mentioned, the correct place would be the sidebar. Once it has been introduced, the support for multiple accounts can be added.
Please be patient and don't expect this feature soon, i have no much spare time, too.

**Comment by schaarsc** [comment]
could  this be addressed with #418 ?

**Comment by stefan-niedermann** [comment]
in theory yes, but i want to keep the steps as small as possible. therefore this is a follow-up to #418

**Comment by mMuck** [comment]
> Hi, is there any chance that you could add support for Multiple NextCloud Servers?
> 
> We are using NextCloud in our company, but I'm also using NextCloud at home and it would be nice to be able to switch between them.

I have the same "issue" as @jkalousek. Were there steps in that direction? (I know, this is developed in freetime. Thanks BTW!!)

**Comment by stefan-niedermann** [comment]
There are some considerations in the Single-Sign-On-library that is currently in development. It would allow to easily maintain multiple accounts.

Currently there are some bugs, we are ironing out while developing a [Deck-client](https://github.com/stefan-niedermann/nextcloud-deck). If the lib is stable, we will consider to implement it in the notes app and finally can get multi-account-support implemented in a nice and reliable way. :)

TL;DR: Currently no direct progress at the notes app, but progress for the whole ecosystem :)

**Comment by DeadSuperHero** [comment]
Thanks for the update, @stefan-niedermann ! 

I just wanted to add a +1 to this, I currently use NextCloud Notes for a publishing organization that I run, and have strongly been considering setting up a second instance for personal use. Being able to connect to multiple accounts on the mobile client would definitely motivate me to move off of the proprietary notes solutions that I currently have on my devices.

**Comment by stefan-niedermann** [comment]
# Early birds attention please :)

@korelstar @dan0xii and me have started some work to implement SSO in notes.

This is a **very early state** which still has some problems left to be resolved. (Mainly regarding migration, widgets and most importantly: The sync always performs a full sync due to a issue in the SSO lib).

If you want, you are welcome to join the work, report bugs and test the branch. We are happy about every feedback. Please provide feedback directly as comment in the PR.

https://github.com/stefan-niedermann/nextcloud-notes/pull/599

**Comment by stefan-niedermann** [comment]
Released on F-Droid (will be there in a few days). Check it out, test it, report bugs!

Thank you for patience.

Play Store will take some more time and more eye for review :)
